test/msd_test.py

```diff
@@ -373,6 +375,63 @@ def test_mass_storage(workspace, parent_test):
 
+    if page_erase_supported:
+        # Test page erase, a.k.a. sector erase by generating iHex with discrete addresses,
+        # programing the device then comparing device memory against expected content.
+        test = MassStorageTester(board, test_info, "Sector Erase")
+        with MbedBoard.chooseBoard(board_id=board.get_unique_id(), init_board=False) as mbed_board:
+            memory_map = mbed_board.target.getMemoryMap()
+        flash_regions = [region for region in memory_map if region.type == 'flash']
+        number_flash_regions = len(flash_regions)
+
+        max_address = intel_hex.maxaddr()
+        # Create an object. We'll add the addresses of unused even blocks to it first, then unused odd blocks for each region
+        ih = intelhex.IntelHex()
+        # Add the content from test bin first
+        expected_bin_contents = bin_file_contents
+        for region_index, the_region in enumerate(flash_regions):
+            flash_start = the_region.start
+            flash_length = the_region.length
+            block_size = the_region.blocksize
+
+            number_of_blocks = flash_length // block_size
+
+            # Sanity check the regions are contiguous
+            if region_index:
+                assert flash_start == (flash_regions[region_index - 1].start + flash_regions[region_index - 1].flash_length)
+
+            if max_address >= (flash_start + flash_length):
+                # This bin image crosses this region, don't modify the content, go to the next region
+                continue
+            elif max_address >= flash_start:
+                # This bin image occupies partial region. Skip the used portion to avoid touching any security bits and pad the rest
+                expected_bin_contents += bytearray([0xff]) * (flash_start + flash_length - max_address - 1)
+                # Calculate the starting block after the image to avoid stumbling upon security bits
+                block_start = (max_address - flash_start) // block_size + 1
+            else:
+                # This bin image doesn't reach this region
+                expected_bin_contents += bytearray([0xff]) * flash_length
+                block_start = 0
+            # For all even blocks, overwrite all addresses with 0x55; for all odd blocks, overwrite all addresses with 0xAA
+            for pass_number in range (2):
+                if pass_number == 0:
+                    modifier = 0x55
+                else:
+                    modifier = 0xAA
+                    block_start += 1
+                for block_idx in range(block_start, number_of_blocks, 2):
+                    for address_to_modify in range (flash_start + block_idx * block_size, flash_start + (block_idx + 1) * block_size):
+                        expected_bin_contents[address_to_modify] = modifier
+                        ih[address_to_modify] = modifier
+        if not os.path.exists("tmp"):
+            os.makedirs("tmp")
```

Comment:
(Not in this PR, but soon…) Take a look at the Use the built-in `tempfile` Python package to create temp files in the system temp directory instead of creating your own local "tmp" dir.

Reply:
Got it, thanks for letting me know.